src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/comments/ScannerWithCommentEnabledTest.kt

```diff
@@ -0,0 +1,363 @@
+
+
+            """.trimIndent()
+        )
+
+        assertTokensEqual(
+            expectedList = expected,
+            expectedScalarValueList = expectedScalarValue,
+            sut = sut,
+        )
+    }
+
+    test("block scalar - keep new line - no new line at and") {
+        val expected = listOf(
+            ID.StreamStart,
+            ID.BlockMappingStart,
+            ID.Key, ID.Scalar, ID.Value, ID.Comment, ID.Scalar,
+            ID.BlockEnd,
+            ID.StreamEnd,
+        )
+        val expectedScalarValue = listOf("abc", "def\nhij")
+
+        val sut = constructScanner(
+            """
+            abc: |- # Comment
+                def
+                hij
+
+
+            """.trimIndent()
+        )
+
+        assertTokensEqual(
+            expectedList = expected,
+            expectedScalarValueList = expectedScalarValue,
+            sut = sut,
+        )
+    }
+
+    test("block scalar - keep new line - all new lines at and") {
+        val expected = listOf(
+            ID.StreamStart,
+            ID.BlockMappingStart,
+            ID.Key, ID.Scalar, ID.Value, ID.Comment, ID.Scalar,
+            ID.BlockEnd,
+            ID.StreamEnd,
+        )
+        val expectedScalarValue = listOf("abc", "def\nhij\n\n")
+
+        val sut = constructScanner(
+            """
+            abc: |+ # Comment
+                def
+                hij
+
+
+            """.trimIndent()
+        )
+
+        assertTokensEqual(
+            expectedList = expected,
+            expectedScalarValueList = expectedScalarValue,
+            sut = sut,
+        )
+    }
+
+    test("directive line end comment") {
+        val expected = listOf(
+            ID.StreamStart,
+            ID.Directive,
+            ID.Comment,
+            ID.StreamEnd,
+        )
+
+        val sut = constructScanner("%YAML 1.1 #Comment\n")
+
+        assertTokensEqual(
+            expectedList = expected,
+            sut = sut,
+        )
+    }
+})
+
+private fun constructScanner(input: String): Scanner {
+    val settings = LoadSettings.builder().setParseComments(true).build()
+    return ScannerImpl(settings, StreamReader(settings, input))
+}
+
+private fun assertTokensEqual(
+    expectedList: List<ID>,
+    expectedScalarValueList: List<String>? = null,
+    sut: Scanner,
+) {
+    val expectedIterator = expectedList.iterator()
+    val expectedScalarValueIterator = expectedScalarValueList?.iterator()
+
+    while (!sut.checkToken(ID.StreamEnd)) {
+        val token = sut.next()
+        assertTokenEquals(
+            expectedIdIterator = expectedIterator,
+            expectedScalarValueIterator = expectedScalarValueIterator,
+            token = token,
+        )
+    }
+
+    val token = sut.peekToken()
+    assertTokenEquals(
+        expectedIdIterator = expectedIterator,
+        expectedScalarValueIterator = expectedScalarValueIterator,
+        token = token,
+    )
+    expectedIterator.hasNext().shouldBeFalse()
```

Comment:
suggestion: I would add a clue here saying `unexpected tokens` or something like that. Right now we will only see `expected: false, actual: true`